frontend/src/components/dashboard/index.js

```diff
@@ -6,9 +6,19 @@ import { useStore } from '../../store'
 import poster from '../../Media/pupper_cool.jpeg'
 import TabPanel from '../tabPanel'
 import InputController from '../inputController'
+import AdditionalActions from '../additionalActions'
+import RobotSpeed from '../robotSpeed'
+import RobotHardware from '../robotHardware'
 
 const Dashboard = observer(() => {
-  const {navigationStore: { activeTab }} = useStore()
+  const {
+    navigationStore: { activeTab },
+    settingsStore: { widgets }
+  } = useStore()
+
+  const isBatteryWidgetSelected = widgets.find((widget) => widget.id === '2').selected
```

Comment:
Filtering by hardcoded string ids might not be reliable. Can we make it more generic?

Reply:
done